src/Raven.Client/Documents/Operations/Revisions/DeleteRevisionsManuallyOperation.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Extensions;
+using Raven.Client.Http;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Operations.Revisions
+{
+    public class DeleteRevisionsManuallyOperation : IOperation<DeleteRevisionsManuallyOperation.Result>
+    {
+        private DeleteRevisionsRequest _request;
+
+        public DeleteRevisionsManuallyOperation(List<string> revisionsChangeVecotors)
+        {
+            if (revisionsChangeVecotors == null)
+            {
+                throw new ArgumentNullException(nameof(revisionsChangeVecotors), "'revisionsChangeVecotors' cannot be both null or empty.");
+            }
+
+            _request = new DeleteRevisionsRequest { RevisionsChangeVecotors = revisionsChangeVecotors };
+        }
+
+        public DeleteRevisionsManuallyOperation(string documentId, long maxDeletes = 1024, DateTime? after = null, DateTime? before = null)
```

Comment:
maximum number of deletes you can do in this operation,
@ayende said that all deletions should be in one transaction, but the number of deletions in a transaction should be limited by the 'maxDeletes' - a number that comes from the client, 
because we shouldn't make a big number of deletions in one transaction.

Reply:
The issue is what do we do when we have a document with more revisions for that in the range?

Reply:
You stop deleting when you exceed the `maxDeletes`.
The client gets the number of deletions in the end, and he can see if it exceeded or not.